newhelm/base_test.py

```diff
@@ -75,6 +69,18 @@ def measure_quality(self, item: TestItemAnnotations) -> Dict[str, float]:
         pass
 
     @abstractmethod
-    def aggregate_measurements(self, items: List[MeasuredTestItem]) -> List[Result]:
-        """Combine the measurements for each TestItem into a list of Results."""
+    def aggregate_measurements(self, items: List[MeasuredTestItem]) -> Typeable:
+        """Combine the measurements for each TestItem into a test specific Typeable."""
         pass
+
+
+class TestResult(TypedData):
```

Comment:
Creating a subclass makes the code a little more explicit and provides some type checking. For example in `TestRecord` we can have `result: TestResult` instead of `result: TypedData`, which means when constructing a `TestRecord` you have to do `TestResult.from_instance(` instead of `TypedData.from_instance`. 

If you think its adding more confusion than it is saving, I'd be fine removing it. Note that `Annotation` is currently the same way, so I'd also remove that if we think `TestRecord` isn't worth it.

Reply:
Lets keep it in that case for the consistency.